lib/ansible/modules/cloud/amazon/iam_group.py

```diff
@@ -0,0 +1,376 @@
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        # Detach policies not present
+        current_attached_policies_arn_list = []
+        for policy in current_attached_policies:
+            current_attached_policies_arn_list.append(policy['PolicyArn'])
+
+        for policy_arn in list(set(current_attached_policies_arn_list) - set(managed_policies)):
+            try:
+                connection.detach_group_policy(GroupName=params['GroupName'], PolicyArn=policy_arn)
+            except (ClientError, ParamValidationError) as e:
+                module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+
+        # If there are policies in managed_policies attach each policy
+        if managed_policies != [None]:
+            for policy_arn in managed_policies:
+                try:
+                    connection.attach_group_policy(GroupName=params['GroupName'], PolicyArn=policy_arn)
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        changed = True
+
+    # Manage group memberships
+    current_group_members = get_group(connection, params['GroupName'])['Users']
+    current_group_members_list = []
+    for member in current_group_members:
+        current_group_members_list.append(member['UserName'])
+
+    if not compare_group_members(current_group_members_list, users):
+
+        # If users has a single empty element we want to remove all users that are members of the group
+        if len(users) == 1 and users[0] is None:
+            for user in current_group_members_list:
+                try:
+                    connection.remove_user_from_group(GroupName=params['GroupName'], UserName=user)
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        for user in list(set(current_group_members_list) - set(users)):
+            try:
+                connection.remove_user_from_group(GroupName=params['GroupName'], UserName=user)
+            except (ClientError, ParamValidationError) as e:
+                module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        if users != [None]:
+            for user in users:
+                try:
+                    connection.add_user_to_group(GroupName=params['GroupName'], UserName=user)
+                except (ClientError, ParamValidationError) as e:
+                    module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        changed = True
+
+    # Get the group again
+    group = get_group(connection, params['GroupName'])
+
+    module.exit_json(changed=changed, iam_group=camel_dict_to_snake_dict(group))
+
+
+def destroy_group(connection, module):
+
+    params = dict()
+    params['GroupName'] = module.params.get('name')
```

Comment:
You don't really seem to be using `params` as a useful dict at all in this method. Setting `group_name = module.params.get('name')` and then using `group_name` rather than `params['GroupName']` would make more sense